onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -3026,10 +3028,54 @@ def aten_imag(self: TensorType) -> TensorType:
     raise NotImplementedError()
 
 
-def aten_index(self: TensorType, indices: Optional[Sequence[TensorType]]) -> TensorType:
+def are_consecutive(lst):
+    if len(lst) == 0:
+        return True
+    else:
+        return sorted(lst) == list(range(min(lst), max(lst) + 1))
+
+
+def _is_none_in_middle(indices):
+    not_none_indices = [i for i, idx in enumerate(indices) if idx is not None]
+    return not are_consecutive(not_none_indices)
+
+
+@torch_op("aten::index", trace_only=True)
+def aten_index(self: TensorType, indices: Sequence[Optional[INT64]]) -> TensorType:
```

Comment:
This should work for all cases except for bool mask index, if possible.

Let me know if you can find a bug! @justinchuby